modules/planning/scenarios/park/valet_parking/stage_approaching_parking_spot.cc

```diff
@@ -20,30 +20,56 @@
 
 #include "modules/planning/scenarios/park/valet_parking/stage_approaching_parking_spot.h"
 
+#include <vector>
+
 #include "modules/common/configs/vehicle_config_helper.h"
 #include "modules/common/vehicle_state/vehicle_state_provider.h"
+#include "modules/map/hdmap/hdmap_util.h"
 
 namespace apollo {
 namespace planning {
 namespace scenario {
 namespace valet_parking {
 
+using apollo::common::PointENU;
+using apollo::hdmap::ParkingSpaceInfoConstPtr;
+
 Stage::StageStatus StageApproachingParkingSpot::Process(
     const common::TrajectoryPoint& planning_init_point, Frame* frame) {
   ADEBUG << "stage: StageApproachingParkingSpot";
   CHECK_NOTNULL(frame);
   GetContext()->target_parking_spot_id.clear();
-  if (frame->local_view().routing->routing_request().has_parking_space() &&
-      frame->local_view().routing->routing_request().parking_space().has_id()) {
-    GetContext()->target_parking_spot_id = frame->local_view()
-                                               .routing->routing_request()
-                                               .parking_space()
-                                               .id()
-                                               .id();
-  } else {
-    AERROR << "No parking space id from routing";
-    return StageStatus::ERROR;
+  if (frame->local_view().routing->routing_request().has_parking_info()) {
```

Comment:
save "rame->local_view().routing->routing_request()" to a const variable.

Reply:
Fixed. Thank you.